sql/core/src/test/resources/sql-tests/results/pgSQL/date.sql.out

```diff
@@ -508,48 +508,128 @@ struct<Days From 2K:int>
 
 
 -- !query 47
-select make_date(2013, 7, 15)
+SELECT DATE_TRUNC('MILLENNIUM', TIMESTAMP '1970-03-20 04:30:00.00000')
 -- !query 47 schema
-struct<make_date(2013, 7, 15):date>
+struct<date_trunc(MILLENNIUM, TIMESTAMP('1970-03-20 04:30:00')):timestamp>
 -- !query 47 output
-2013-07-15
+1001-01-01 00:07:02
```

Comment:
Non-zero time of `07:02` is due to incompatibility in calculations of time zone offsets by old API `java.sql.Date` and new one from Java 8 time API, see https://bugs.openjdk.java.net/browse/JDK-6281408.

To get correct result, set `spark.sql.datetime.java8API.enabled` to `true`.

Reply:
shall we at least enable it in `Dataset.show`?

Reply:
I think so but I would try that separately because it can require changes not related to truncations.

Reply:
Here is the JIRA ticket: https://issues.apache.org/jira/browse/SPARK-28643

Reply:
cool, thanks!